File: Chain of Responsibility Pattern/GeneralDoctor.java
package generaldoctor;

import patient.Patient;
import diseaseseverity.DiseaseSeverity;
import handler.Handler;

public class GeneralDoctor extends Handler{
	
	@Override
	public void handlePatient(Patient patient){
		
		if(patient.getDiseaseSeverity() == DiseaseSeverity.MODERATE){
			System.out.println("Diagnose the Patient " + patient.getName() + " by the General Doctor \n");
		}else{
			System.out.println("Passing the Patient " + patient.getName() + " to Higher Authority from General Doctor");
			this.successor.handlePatient(patient);
		}
		
	}
	
}

File: Design Patterns/Chain of Responsibility Pattern in JavaScript/GeneralDoctor.js
const Handler = require('./Handler');
const DiseaseSeverity = require('./DiseaseSeverity');

class GeneralDoctor extends Handler {

    handlePatient(patient) {

        super.handlePatient(patient);

        if (patient.diseaseSeverity === DiseaseSeverity.MODERATE) {
            console.log("Diagnose the Patient " + patient.name + " by the General Doctor \n");
        } else {
            console.log("Passing the Patient " + patient.name + " to Higher Authority from General Doctor");
            this.successor.handlePatient(patient);
        }

    }

}

module.exports = GeneralDoctor;